### DISCUSSION TITLE
Z-Stop for BLTouch, bed size, baud rate, etc

### DISCUSSION THREAD

**Discussion by d3relict** [Jyers/Marlin#412] [Ideas]
Hi,

tried to RTFM, but did not find what i was looking for.

coming from Smith3D firmware for Ender 3 V2, i'm currently using a version that has Z-Stop switch for BLTouch, sets a 230x230mm bed, 5x5 fast leveling, and 250000 baud rate. as far as i seen - based on file names -, the release bin files do not offer all of these.

@Jyers can you please confirm if these feature are available in your binaries, or do you plan to add them any time soon?

thanks!

**Comment by Jyers** [comment]
All binaries have a bed size of 230x230, the BLTOUCH-5x5-HS label indicates a 5x5 high speed leveling grid. However all binaries use the dedicated BL Touch port and 115200 baud rate. Unfortunately variations including different port connections will most likely not be made to avoid overloading my releases with too many binaries. I may change the default baud rate at some point, but at the moment I have not done enough research to understand the possible pros and cons. If you need these features I would highly recommend looking into compiling your own version of the firmware as it gives you much greater control over the enabled features and settings.

**Reply by d3relict** [reply]
just a note: when using the dedicated bltouch port, i had issues when the touch sensor was ignored, and the nozzle hit the bed. since i'm using the z-stop port, such issues are gone. please consider, in the mean time i'll look into compiling my own. thanks!

**Reply by Jyers** [reply]
Yes, there seem to be a lot of BL Touch connectivity issues going around. The issue is while some probes ship with two connectors that allow it to be used in the endstop or dedicated port, some ship with one connector that can only be used in the BL Touch port. This is why the default connection is the dedicated port for this firmware. Have you seen the [resistor fix](https://user-images.githubusercontent.com/80574048/111040575-6cdfd280-842b-11eb-971c-446933b9b475.jpeg)? If you're handy with a soldering iron it could save you some compiling. Even though I always recommend compiling your own firmware for the added customizability.

**Reply by VashBaldeus** [reply]
> 
> 
> Yes, there seem to be a lot of BL Touch connectivity issues going around. The issue is while some probes ship with two connectors that allow it to be used in the endstop or dedicated port, some ship with one connector that can only be used in the BL Touch port. This is why the default connection is the dedicated port for this firmware. Have you seen the [resistor fix](https://user-images.githubusercontent.com/80574048/111040575-6cdfd280-842b-11eb-971c-446933b9b475.jpeg)? If you're handy with a soldering iron it could save you some compiling. Even though I always recommend compiling your own firmware for the added customizability.

As far as I figured it out, BLTouch 3.1 ships only for the 32bit boards, anything earlier ships with 2 connectors. At least from what I've seen, that any earlier version of the BLTouch prior 3.1 ships with older style connector.

**Reply by mmithril** [reply]
I went through several threads and it seems the resistor fix sometimes helps and sometimes does not while the using Z port always helps.

It is quite easy to use a e.g. a needle to detach 2 pins both from the Z switch cable connector and 2 from the BLtouch connector and just swap them to use the Z switch port even with the new cable.

![obrázok](https://user-images.githubusercontent.com/11449451/175479905-2cd01abc-5b28-4e92-b224-6958c624c8fe.png)
(from https://www.smith3d.com/2020/08/01/ender-3-v2-bltouch-firmware-installation-guide-by-smith3d-com/ section _Sometimes Probe doesn’t finish during leveling_)

But be careful - my BLTouch came with a cable having the same colors but in reversed order!
